### DISCUSSION TITLE
Merge BAC_DataSource.ObjectName and BAC_DataSource.SQLStatement into BAC_DataSource.RecordSource?

### DISCUSSION THREAD

**Discussion by josef-poetzl** [team-moeller/better-access-charts#118] [General]
Do the properties ObjectName  and SQLStatement have a reason to be kept separate?
I think it would be easier to manage if only one property (similar to Form.RecordSource) is used.

Example: BAC_DataSource.getData
```vba
'Detect data source
    Select Case m_Type
        Case dstTableName, dstQueryName:
            Set rst = db.OpenRecordset(ObjectName)
        Case dstSQLStament:
            Set rst = db.OpenRecordset(SQLStatement)
        Case dstRecordset:
            Set rst = Me.Recordset
        ...
```
vs.
```vba
'Detect data source
    Select Case m_Type
        Case dstTableName, dstQueryName, dstSQLStament:
            Set rst = db.OpenRecordset(RecordSource)
        Case dstRecordset:
            Set rst = Me.Recordset
        ...
```

**Comment by team-moeller** [comment]
Let's see how this looks in the users code:
`myChart.Chart.DataSource.RecordSource= "tbl_DemoData"`
or
`myChart.Chart.DataSource.RecordSource= "qry_DemoData"`
or
`myChart.Chart.DataSource.RecordSource= "Select * from tbl_DemoData order by pkid desc"`

Is this intuitive?

**Comment by josef-poetzl** [comment]
I think it's consistent.

Examples:

* RecordSource 
```vba
Dim frm as Access.Form
Set frm = ..
frm.RecordSource = "tbl_DemoData"
frm.RecordSource = "qry_DemoData"
frm.RecordSource = "Select * from tbl_DemoData order by pkid desc"
```

* RowSource instead of RecordSource 
```vba
Dim ctl as Access.Combobox  ' or Listbox
Set ctl = ..
ctl.RowSource = "tbl_DemoData"
ctl.RowSource = "qry_DemoData"
ctl.RowSource = "Select * from tbl_DemoData order by pkid desc"
```
Access.Chart also has the RowSource property.
Maybe RowSource is better (more intuitive) than RecordSource.

My main reason: 
```myChart.Chart.DataSource.ObjectName = "Select * from tbl_DemoData order by pkid desc"```
will run without errors since SQLStatement and ObjectName do the same thing. If two properties can be used equally, then why not just use one property?

**Comment by team-moeller** [comment]
OK, good points.

In Access VBA your code looks like this:
`Access.Chart.Rowsource = ...`

With better Access Chart your code would look like this:
`myChart.Chart.DataSource.Rowsource = ...`

That sounds weird to me.

**Comment by josef-poetzl** [comment]
Then let's find another name for Rowsource.

ADODB.Command uses CommandText. The content can be an SQL statement or just a procedure name (Depending on CommandType property).
=> ```myChart.Chart.DataSource.DataSourceText``` ... that doesn't sound much better to me - but maybe better than RowSource ;-)

Other (also not much better) names:
```myChart.Chart.DataSource.Text``` ... missing the exact description of the content here
```myChart.Chart.DataSource.Value``` ... the same here


Maybe with default property in DataSource class:
```
myChart.Chart.DataSource = "TableDefName"
'or
myChart.Chart.DataSource = "select ... from ..."
' other settings:
myChart.Chart.DataSource.LabelFieldName = ...
...
```

Object-oriented approach
```
Set myChart.Chart.DataSource = New SqlDataSource(...)
Set myChart.Chart.DataSource = New TableQueryNameDataSource(...)
Set myChart.Chart.DataSource = New RecordsetDataSource(...)
Set myChart.Chart.DataSource = New DatasetDataSource(...)
Set myChart.Chart.DataSource = New XmlDataSource(...)
Set myChart.Chart.DataSource = New JsonDataSource(...)
Set myChart.Chart.DataSource = New CsvDataSource(...)
```
But we're using VBA, so I'll drop that thought right away.


[OT]
The good news:  In my fork, the user of the library no longer has to set the data source with VBA code. Then you can no longer see the name. :-)

**Comment by team-moeller** [comment]
I think the default property approach is the best idea here.
Will think about this and come back when I have the head free. At the moment I have too many things going on.
